src/main/java/com/google/cloud/resourcemanager/ResourceManagerImpl.java

```diff
@@ -273,6 +273,141 @@ public Policy replacePolicy(final String projectId, final Policy newPolicy) {
+          runWithRetries(
+              new Callable<com.google.api.services.cloudresourcemanager.model.Lien>() {
+                @Override
+                public com.google.api.services.cloudresourcemanager.model.Lien call() {
+                  return resourceManagerRpc.getLien(name);
+                }
+              },
+              getOptions().getRetrySettings(),
+              EXCEPTION_HANDLER,
+              getOptions().getClock());
+      return answer == null ? null : Lien.fromPb(this, answer);
+    } catch (RetryHelperException ex) {
+      throw ResourceManagerException.translateAndThrow(ex);
+    }
+  }
+
+  private static class LienPageFetcher implements NextPageFetcher<Lien> {
+
+    private static final long serialVersionUID = 2158209410430566961L;
+    private final Map<ResourceManagerRpc.Option, ?> requestOptions;
+    private final ResourceManagerOptions serviceOptions;
+    private final String parent;
+
+    LienPageFetcher(
+        String parent,
+        ResourceManagerOptions serviceOptions,
+        String cursor,
+        Map<ResourceManagerRpc.Option, ?> optionMap) {
+      this.requestOptions =
+          PageImpl.nextRequestOptions(ResourceManagerRpc.Option.PAGE_TOKEN, cursor, optionMap);
+      this.serviceOptions = serviceOptions;
+      this.parent = parent;
+    }
+
+    @Override
+    public Page<Lien> getNextPage() {
+      return listLiens(parent, serviceOptions, requestOptions);
+    }
+  }
+
+  @Override
+  public Page<Lien> listLiens(String parent, LienListOption... options) {
+    return listLiens(parent, getOptions(), optionMap(options));
+  }
+
+  private static Page<Lien> listLiens(
+      final String parent,
+      final ResourceManagerOptions serviceOptions,
+      final Map<ResourceManagerRpc.Option, ?> optionsMap) {
+    try {
+      Tuple<String, Iterable<com.google.api.services.cloudresourcemanager.model.Lien>> result =
+          runWithRetries(
+              new Callable<
+                  Tuple<
+                      String,
+                      Iterable<com.google.api.services.cloudresourcemanager.model.Lien>>>() {
+                @Override
+                public Tuple<
+                        String, Iterable<com.google.api.services.cloudresourcemanager.model.Lien>>
+                    call() {
+                  return serviceOptions
+                      .getResourceManagerRpcV1Beta1()
+                      .listLiens(parent, optionsMap);
+                }
+              },
+              serviceOptions.getRetrySettings(),
+              EXCEPTION_HANDLER,
+              serviceOptions.getClock());
+      String cursor = result.x();
+
+      Iterable<Lien> liens =
+          result.y() == null
+              ? ImmutableList.<Lien>of()
+              : Iterables.transform(
+                  result.y(),
+                  new Function<com.google.api.services.cloudresourcemanager.model.Lien, Lien>() {
+                    @Override
+                    public Lien apply(
```

Comment:
I'm concerned that there are two different classes named Lien/ Could one be renamed?

Reply:
done